File: data-munging/jakejgordon/katas/AdderKata/AdderTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;

namespace katas.AdderKata
{
    [TestFixture]
    public class AdderTests
    {
        private Adder adder;

        [SetUp]
        public void SetUp()
        {
            adder = new Adder();
        }

        [Test]
        public void ItReturnsZeroIfThereIsOnlyAnEmptyString()
        {
            var result = adder.Add(string.Empty);

            Assert.That(result, Is.EqualTo(0));
        }

        [TestCase("0")]
        [TestCase("1")]
        [TestCase("2")]
        [TestCase("99")]
        public void ItReturnsTheNumberThatWasPassedInIfThereIsOnlyOneNumber(string theNumberPassedIn)
        {
            var result = adder.Add(theNumberPassedIn);

            Assert.That(result, Is.EqualTo(int.Parse(theNumberPassedIn)));
        }
    }
}

File: adder_new_lang/dougmorgan/StringCalculator/Adder/AdderTests.fs
module AdderTests

    open NUnit.Framework
    open FsUnit
    open StringCalculator.Adder

    [<TestFixture>]
    type AdderTests() = class

        [<Test>]
        member x. ``should return zero when for empty string``() = 
            Assert.That(Add(""), Is.EqualTo(0))

        [<Test>]
        member x. ``should throw exception if negative number included``() = 
         Assert.Throws<System.ArgumentException>(fun () -> 
            Add("1,-2") |> ignore) |> ignore

        [<Test>]
        member x. ``should not include numbers greater than or equal to 1000``() = 
            Assert.That(Add("5,1000"), Is.EqualTo(5))

        [<TestCase("1",1)>]
        [<TestCase("5",5)>]
        member x. ``should sum up single addend" ``(addend:string, expectedSum:int) =
            let actualSum = Add(addend)
            Assert.That(actualSum, Is.EqualTo(expectedSum))

        [<TestCase("6,7,8",21)>]
        [<TestCase("6\n7\n8",21)>]
        [<TestCase("//;\n6;7;8",21)>]
        member x. ``should sum up multiple addend" ``(addends:string, expectedSum:int32) = 
            let actualSum = Add(addends)
            Assert.That(actualSum, Is.EqualTo(expectedSum))
    end